### ISSUE TITLE
CMake does not rebuild simint if vectorization is mismatched

### ISSUE THREAD

**Issue by TiborGY** [psi4/psi4#2480]
As far as I can tell, the current CMake does not check if the vectorization type (SSE, AVX, etc) of the simint binary found matches the requested simint vectorization type. This shows up by default, as simint from the conda dev env comes with SSE, while the CMake default for simint vectorization is AVX:
```
-- Setting (unspecified) option psi4_CXX_STANDARD: 17
-- Setting (unspecified) option SIMINT_VECTOR: avx
-- Setting (unspecified) option SPHINX_THEME: sphinx_psi_theme
...
-- Found qcengine: /home/gytibor/.conda/envs/p4dev/lib/python3.9/site-packages/qcengine (found version v0.23.0)
-- Found simint 7: /home/gytibor/.conda/envs/p4dev/lib/libsimint.so (found version 0.7; vectorization sse)
-- Found Libxc: /home/gytibor/.conda/envs/p4dev/lib/libxc.so.9 (found version 5.1.5)
```

**Comment by TiborGY** [comment]
IMO the fix would be to rebuild simint if vectorization is mismatched, and either rebuild the conda dev library with AVX or set the CMake default to SSE.

**Comment by loriab** [comment]
You're right, the handling of simint vec type is unsophisticated. The mention of `SIMINT_VECTOR` in psi4's cmake is as indication that if you set the var for psi4, then it'll be passed on to the simint build, should simint be built from source. In that way, something like the below should do what you want. (The second is a standard CMake variable that means don't-look-for-prebuilt.)

```
-D SIMINT_VECTOR=avx
-D CMAKE_DISABLE_FIND_PACKAGE_simint=ON
```

simint itself doesn't offer any handles for choosing an installation of a particular vectorization, nor does it provide any is-this-installation-compatible-with-given-vec handles. also, it's hard to know what the user wants: on an AVX computer (cmake can't quickly say that a computer is AVX, btw), a user getting a sse simint might be quite content b/c it'll run or discontent b/c it's unoptimized. Between the uncertainty and lack of tools, I went with the dumb handling you observed. For background, the conda package is sse  so that it runs everywhere. And the cmake vector variable default is AVX since that's a likely minimum, and a wrong choice produces only local trouble.

I can see that the top-level CMakeList.txt text needs to be clearer so the vector status isn't surprising. But I'm not sure if more logic is worth it. What do you think?
